boa_engine/src/vm/code_block.rs

```diff
@@ -1256,14 +1241,22 @@ impl JsObject {
 
         context.vm.push_frame(frame);
 
+        // Push return value.
+        context.vm.push(JsValue::undefined());
```

Comment:
Would it make sense to tie pushing the return/completion value to pushing the frame onto the stack?

```rust
pub(crate) fn push_frame(&mut self, mut frame: CallFrame) {
    let current_stack_length = self.stack.len();
    frame.set_frame_pointer(current_stack_length as u32);

    self.push(JsValue::undefined());
    self.frames.push(frame);
}
```

Reply:
> Would it make sense to tie pushing the return/completion value to pushing the frame onto the stack?

Yes, thought about this, for normal function this is fine, the issue is with generators since we need to restore without adding anything.

One solution would be to have two `push_frame` functions, or maybe extract the parts into a `prepare_call_frame()`.